frontend/test/pytest/test_gradient.py

```diff
@@ -1620,13 +1622,15 @@ def body(i, result_array):
 
         return body(transposed_result)
 
-    cat_res = qjit(
+    cat_res_func = qjit(
         jacobian(
             my_model,
             argnums=1,
         ),
         keep_intermediate=True,
-    )(data, params["weights"])
+    )
+    shutil.rmtree("grad.my_model")
```

Comment:
Do you think we could find a way to avoid hard-coding the name `"grad.my_model"`? This is probably unlikely, but suppose we change the name of the `my_model` function, `shutil.rmtree` would raise a `FileNotFoundError` exception. Maybe something like

```python
shutil.rmtree(f"grad.{my_model.__name__}")
```

Another option might just be check if the directory exists first, e.g.

```python
tmp_dir = "grad.my_model"
if os.path.isdir(tmp_dir):
    shutil.rmtree(tmp_dir)
```

Reply:
```shutil.rmtree(f"grad.{my_model.__name__}")``` will fail too if the function `my_model` changes to a new name

I think the check is nice though.

Reply:
True, although you'd be trading a `FileNotFoundError` for a `NameError`, but it's hard to say which is a better error to have :wink:. The check-if-exists option is probably the better one then.

Reply:
done

Reply:
Looks good, thanks!